iceoryx_hoofs/source/posix_wrapper/signal_watcher.cpp

```diff
@@ -0,0 +1,74 @@
+#include "iceoryx_hoofs/posix_wrapper/signal_watcher.hpp"
+#include "iceoryx_hoofs/cxx/helplets.hpp"
+
+
+namespace iox
+{
+namespace posix
+{
+void internalSignalHandler(int) noexcept
+{
+    auto& instance = SignalWatcher::getInstance();
+    instance.m_hasSignalOccurred.store(true);
+
+    for (uint64_t i = 0, currentNumberOfWaiter = instance.m_numberOfWaiters.load(); i < currentNumberOfWaiter; ++i)
+    {
+        instance.m_semaphore.post().or_else([](auto) {
+            std::cerr << "Unable to increment semaphore in signal handler" << std::endl;
+            constexpr bool UNABLE_TO_INCREMENT_SEMAPHORE_IN_SIGNAL_HANDLER = false;
+            cxx::Ensures(UNABLE_TO_INCREMENT_SEMAPHORE_IN_SIGNAL_HANDLER);
+        });
+    }
+}
+
+SignalWatcher::SignalWatcher() noexcept
+    : m_semaphore{std::move(Semaphore::create(CreateUnnamedSingleProcessSemaphore, 0U)
+                                .or_else([](auto) {
+                                    std::cerr << "Unable to create semaphore for signal watcher" << std::endl;
+                                    constexpr bool UNABLE_TO_CREATE_SEMAPHORE_FOR_SIGNAL_WATCHER = false;
+                                    cxx::Ensures(UNABLE_TO_CREATE_SEMAPHORE_FOR_SIGNAL_WATCHER);
+                                })
```

Comment:
This would be so much nicer if we would have a `Semaphore::create(...).expects("Unable to create semaphore for signal watcher");` with `expects(...)` doing exactly what you did

Reply:
Mmmhhh ... this we can actually implement. Maybe something for a long christmas night - I'll note that. See #996